presto-native-execution/src/test/java/com/facebook/presto/nativeworker/SymlinkManifestGeneratorUtils.java

```diff
@@ -0,0 +1,151 @@
+        try (BufferedWriter writer = Files.newBufferedWriter(manifestFilePath, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)) {
+            List<String> fileList = new ArrayList<>();
+            collectDataFiles(hiveTableLocation, fileList);
+
+            for (String filePath : fileList) {
+                writer.write(filePath);
+                writer.write(System.lineSeparator());
+            }
+        }
+    }
+
+    /**
+     * Recursively collects all non-hidden file paths in the source directory.
+     * If a directory is encountered, the method is called recursively to collect files in subdirectories.
+     *
+     * @param sourceDir the directory to start collecting files from
+     * @param fileList the list where the collected file paths are stored
+     * @throws IOException if an I/O error occurs while accessing the directory or files
+     */
+    private static void collectDataFiles(Path sourceDir, List<String> fileList) throws IOException
+    {
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(sourceDir)) {
+            for (Path entry : stream) {
+                if (!Files.isHidden(entry)) {
+                    if (Files.isDirectory(entry)) {
+                        collectDataFiles(entry, fileList);
+                    }
+                    else {
+                        fileList.add(entry.toString());
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Cleans up (deletes) all files and subdirectories inside the specified directory.
+     * This method walks through the directory tree and deletes files in reverse order to ensure
+     * that directories are deleted after their contents.
+     *
+     * @param directory the root directory to clean up
+     * @throws IOException if an I/O error occurs during file deletion
+     */
+
+    public static void cleanupSymlinkData(Path directory) throws IOException
+    {
+        MoreFiles.deleteRecursively(directory, RecursiveDeleteOption.ALLOW_INSECURE);
+    }
+
+    /**
+     * Returns the Hive type string corresponding to a given TPCH type.
+     * Currently only supports conversion for "integer" TPCH type.
+     *
+     * @param tpchType the TPCH type as a string
+     * @return the corresponding Hive type as a string
+     */
+    public static String getHiveTypeString(String tpchType)
+    {
+        switch (tpchType) {
+            case "integer":
+                return "int";
+            default:
+                return tpchType;
+        }
+    }
+
+    /**
+     * Retrieves a list of columns for a specified TPCH table.
+     * The method looks up the TPCH table by name and converts its columns to Hive-compatible
+     * column representations.
+     *
+     * @param tableName the name of the TPCH table
+     * @return a list of Column objects representing the columns of the table
+     */
+    public static List<Column> getTpchTableColumns(String tableName)
```

Comment:
Don't understand why this method is part of SymlinkManifestGeneratorUtils. You could add it as a private method in AbstractTestNativeHiveExternalTableTpchQueries.java itself.

Reply:
Though this method is not specific to symlink tables but is generic if we need to get columns list for a given tpch table using TPCH API. Which is why I kept it in Utils.

If you think we should move it to `AbstractTestNativeHiveExternalTableTpchQueries` I can do that, LMK?

Reply:
I feel its better in AbstractTestNativeHiveExternalTableTpchQueries.